workers/unity/Assets/Fps/Scripts/Editor/MapBuilderInspector.cs

```diff
@@ -1,37 +1,60 @@
 using UnityEditor;
 using UnityEngine;
-
-[CustomEditor(typeof(MapBuilder))]
-public class MapBuilderInspector : Editor
+namespace Fps
 {
-    public override void OnInspectorGUI()
+    [CustomEditor(typeof(MapBuilder))]
+    public class MapBuilderInspector : Editor
     {
-        var myTarget = (MapBuilder) target;
+        private const int WarnLayersThreshold = 25;
+
+        public override void OnInspectorGUI()
+        {
+            var myTarget = (MapBuilder) target;
 
-        myTarget.Layers = EditorGUILayout.IntField(new GUIContent(
-                "Number of Tile Layers",
-                "N layers corresponds to 4*(N^2) tiles."),
-            myTarget.Layers);
+            myTarget.Layers = Mathf.Max(0,
+                EditorGUILayout.IntField(new GUIContent(
+                    "Number of Tile Layers",
+                    "N layers corresponds to 4*(N^2) tiles."), myTarget.Layers));
 
-        myTarget.Seed = EditorGUILayout.TextField(new GUIContent(
-                "Seed for Map Generator",
-                "Different seeds produce different maps."),
-            myTarget.Seed);
+            var numTiles = GetTotalTilesFromLayers(myTarget.Layers);
 
-        myTarget.EmptyTileChance = EditorGUILayout.FloatField(new GUIContent(
-                "Chance of Empty Tile",
-                "The chance that a tile in one grid square of the world will be empty."),
-            myTarget.EmptyTileChance);
+            GUI.color = myTarget.Layers < WarnLayersThreshold ? Color.white : Color.yellow;
+            GUILayout.Label($"Number of tile to generate: {numTiles}");
+            GUI.color = Color.white;
 
-        if (GUILayout.Button("Generate Map"))
-        {
-            myTarget.Clean();
-            myTarget.Build();
+            myTarget.Seed = EditorGUILayout.TextField(new GUIContent(
+                    "Seed for Map Generator",
+                    "Different seeds produce different maps."),
+                myTarget.Seed);
+
+            myTarget.EmptyTileChance = EditorGUILayout.FloatField(new GUIContent(
+                    "Chance of Empty Tile",
+                    "The chance that a tile in one grid square of the world will be empty."),
+                myTarget.EmptyTileChance);
+
+            if (GUILayout.Button("Generate Map"))
+            {
+                if (myTarget.Layers < WarnLayersThreshold
+                    || EditorUtility.DisplayDialog("Generate Map Confirmation",
```

Comment:
It may be easier to read to factor this out to a method:


```
private bool GetUserConfirmation()
{
	return EditorUtility.DisplayDialog(...)
}
```

So we could just have `if (myTargetLayers < WarnLayersThreshold || GetUserConfirmation())`